base/array.jl

```diff
@@ -836,9 +839,10 @@ function deleteat!(a::Vector, inds)
     return a
 end
 
+# Simpler and more efficient version for logical indexing
 function deleteat!(a::Vector, inds::AbstractVector{Bool})
     n = length(a)
-    length(inds) == n || throw(BoundsError(a, inds))
+    @boundscheck length(inds) == n || throw(BoundsError(a, inds))
     p = 1
     for (q, i) in enumerate(inds)
```

Comment:
This is an interesting case to discuss the `@inbounds` policy (https://github.com/JuliaLang/julia/issues/20469): am I allowed to move the `@inbounds` before the `for` here? That would allow getting rid of one bounds check per iteration.